src/platform/backends/qemu/iptables_config.cpp

```diff
@@ -0,0 +1,269 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "iptables_config.h"
+
+#include <multipass/format.h>
+#include <multipass/logging/log.h>
+#include <multipass/process.h>
+#include <shared/linux/process_factory.h>
+
+namespace mp = multipass;
+namespace mpl = multipass::logging;
+
+namespace
+{
+// QString constants for all if the different iptables calls
+const QString iptables{QStringLiteral("iptables")};
+const QString exclamation{QStringLiteral("!")};
+
+//   Different tables to use
+const QString filter{QStringLiteral("filter")};
+const QString nat{QStringLiteral("nat")};
+const QString mangle{QStringLiteral("mangle")};
+
+//   Chain constants
+const QString INPUT{QStringLiteral("INPUT")};
+const QString OUTPUT{QStringLiteral("OUTPUT")};
+const QString POSTROUTING{QStringLiteral("POSTROUTING")};
+const QString FORWARD{QStringLiteral("FORWARD")};
+
+//   option constants
+const QString dash_C{QStringLiteral("-C")};
+const QString dash_d{QStringLiteral("-d")};
+const QString dash_D{QStringLiteral("-D")};
+const QString dash_i{QStringLiteral("-i")};
+const QString dash_I{QStringLiteral("-I")};
+const QString dash_j{QStringLiteral("-j")};
+const QString dash_m{QStringLiteral("-m")};
+const QString dash_o{QStringLiteral("-o")};
+const QString dash_p{QStringLiteral("-p")};
+const QString dash_s{QStringLiteral("-s")};
+const QString dash_S{QStringLiteral("-S")};
+const QString dash_t{QStringLiteral("-t")};
+const QString dash_w{QStringLiteral("-w")};
+
+//   protocol constants
+const QString udp{QStringLiteral("udp")};
+const QString tcp{QStringLiteral("tcp")};
+
+//   port options and constants
+const QString dport{QStringLiteral("--dport")};
+const QString sport{QStringLiteral("--sport")};
+const QString to_ports{QStringLiteral("--to-ports")};
+const QString port_53{QStringLiteral("53")};
+const QString port_67{QStringLiteral("67")};
+const QString port_68{QStringLiteral("68")};
+const QString port_range{QStringLiteral("1024-65535")};
+
+//   rule target constants
+const QString ACCEPT{QStringLiteral("ACCEPT")};
+const QString MASQUERADE{QStringLiteral("MASQUERADE")};
+const QString REJECT{QStringLiteral("REJECT")};
+const QString RETURN{QStringLiteral("RETURN")};
+
+//   reject rule constants
+const QString reject_with{QStringLiteral("--reject-with")};
+const QString icmp_port_unreachable{QStringLiteral("icmp-port-unreachable")};
+
+auto multipass_iptables_comment(const QString& bridge_name)
+{
+    return QString("generated for Multipass network %1").arg(bridge_name);
+}
+
+auto iptables_rule_exists(const QString& table, const QStringList& rule)
+{
+    auto process =
+        mp::ProcessFactory::instance().create_process(iptables, QStringList() << dash_t << table << dash_C << rule);
```

Comment:
I changed this in appropriate places.  One place I did not is for the `delete_iptables_rule` function since we are just passing in the whole string as captured by `get_iptables_rules`.

Reply:
Sure, makes sense. The renaming helps, thanks!